pyscript.core/docs/README.md

```diff
@@ -0,0 +1,341 @@
+
+```html
+<script type="micropython">
+    def print_type(event, double):
+        # logs "click 4"
+        print(f"{event.type} {double(2)}")
+</script>
+<button micropython-click="print_type(event, lambda x: x * 2)">
+    print type
+</button>
+```
+
+If this example felt a bit verbose, be ensured custom types would work the same:
+
+```html
+<!-- ℹ️ - requires py-script custom type -->
+<button py-click="print(event.type)">
+    print type
+</button>
+```
+
+What is important to understand about *events* in PyScript is that the text within the attribute is executed just like any other inline or external content is, through the very same *interpreter*, with the notably extra feature that the `event` reference is made temporarily available as *global* by *core*.
+
+This really reflects how otherwise native Web inline events handlers work and we think it's a great feature to support ... *but*:
+
+ * if your script runs *asynchronously* the `event` might be gone on the main / UI thread and by that time any of its `event.stopPropagation()` or `event.preventDefault()` goodness will be problematic, as too late to be executed
+ * if your *interpreter* is *experimental*, or incapable of running *synchronous* events, the `event` reference might be less useful
+
+ℹ️ - Please note that if your code runs via *XWorker*, hence in a different thread, there are different caveats and constraints to consider. Please read the [XWorker](#xworker) dedicated section to know more.
+
+#### The type-env
+
+As `env` as attribute has been mentioned already, it is also possible to confine events to a specific *env* by using the *interpreter* or *custom type* name, and followed by `-env`, for the attribute value, providing the desired *env* to interpret and execute the listener.
+
+```html
+<script type="micropython">
+    def log():
+        print(1)
+</script>
+<!-- note the env value -->
+<script type="micropython" env="two">
+    # the button will log 2
+    def log():
+        print(2)
+</script>
+<!-- note the micropython-env value -->
+<button
+    micropython-env="two"
+    micropython-click="log()"
+>
+    log
+</button>
+```
+
+As mentioned before, this will work with `py-env` too, or any custom type defined out there.
+
+
+## XWorker
+
+Whenever computing relatively expensive stuff, such as a *matplot* image, or literally anything else that would take more than let's say 100ms to answer, running your *interpreter* of choice within a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API) is likely desirable, so that the main / UI thread won't block users' actions, listeners, or any other computation going on in these days highly dynamic pages.
+
+However, the native `Worker` utility is incapable of parsing anything that is not *JS* so that we thought that providing a *cross-worker* utility, such as `XWorker` is, would be a great addiction to what *core* offers out of the box.
```

Comment:
```suggestion
However, the native `Worker` utility is incapable of parsing anything that is not *JS* so that we thought that providing a *cross-worker* utility, such as `XWorker` is, would be a great addition to what *core* offers out of the box.
```